shell/renderer/content_settings_observer.cc

```diff
@@ -20,6 +21,11 @@ ContentSettingsObserver::ContentSettingsObserver(
 ContentSettingsObserver::~ContentSettingsObserver() = default;
 
 bool ContentSettingsObserver::AllowDatabase() {
+  if (!base::CommandLine::ForCurrentProcess()->HasSwitch(
+          switches::kEnableWebSQL)) {
+    return false;
+  }
```

Comment:
Is this a purely renderer-side check? Is there any way we can disable this browser-side as well?

Reply:
We can't disable on browser side, it only manages quota for storage and establishes connection to SQLite https://source.chromium.org/chromium/chromium/src/+/master:storage/browser/database/

The renderer side check is pretty consistent and powerful.

There is flag `--disable-databases` https://source.chromium.org/chromium/chromium/src/+/master:content/public/common/content_switches.cc;l=116 but it would disable both indexedDB and webSQL